doc/Makefile

```diff
@@ -0,0 +1,178 @@
+# Makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line.
+SPHINXOPTS    =
```

Comment:
We've found it very useful to bump sphinx warnings to errors, so we add the `-W` option here. This usually prevent silent mistakes from going into master and then cascading into major annoyances later on.